lib/features/digital_guide_view/data/repository/digital_guide_repository.dart

```diff
@@ -0,0 +1,54 @@
+import "package:dio/dio.dart";
+import "package:flutter/foundation.dart";
+import "package:flutter_riverpod/flutter_riverpod.dart";
+import "package:riverpod_annotation/riverpod_annotation.dart";
+
+import "../../../../api_base_rest/client/dio_client.dart";
+import "../../../../config/env.dart";
+import "../models/digital_guide_response.dart";
+
+part "digital_guide_repository.g.dart";
+
+@riverpod
+
+@riverpod
+Future<DigitalGuideResponse> getDigitalGuideData(Ref ref, int id) async {
+
+  debugPrint("start fetching digital guide data");
+
+  try {
+
+    final digitalGuideUrl = "${Env.digitalGuideUrl}/buildings/$id";
+    debugPrint("Url: $digitalGuideUrl");
+    debugPrint("Authorization token: ${Env.digitalGuideAuthorizationToken}");
+    final dio = ref.read(restClientProvider);
+    dio.options.headers["Authorization"] = "Token ${Env.digitalGuideAuthorizationToken}";
+    final response = await dio.get(digitalGuideUrl);
+
+    debugPrint("Response received: ${response.headers.isEmpty}");
+
+    debugPrint("response data: ${response.data}");
+
+    final DigitalGuideResponse digitalGuideResponse = 
```

Comment:
[nitpicking]: imo no need to assign it to variable before returning. Just `return DigitalGuideResponse.fromJson(response.data as Map<String, dynamic>)` would be clear and fine too? what you think